skyreader/utils/handle_map_data.py

```diff
@@ -132,13 +150,40 @@ def extract_map(
         grid_value[np.isnan(grid_value)] = min_map
         grid_value = grid_value.clip(min_map, None)
 
-    return grid_value, grid_ra, grid_dec, equatorial_map
+    if return_uniqs:
+        return grid_value, grid_ra, grid_dec, equatorial_map, uniq_array
+    else:
+        return grid_value, grid_ra, grid_dec, equatorial_map
+
+
+def _fill_first_nside_empty(
+    nside: int,
+    result: SkyScanResult,
+    grid_map: dict,
+    uniq_list: list,
+):
+    """
+    Fill the grid_map at the first nside with empty pixels
+    """
+    tot_npix = healpy.nside2npix(nside)
+    if tot_npix > len(result.get_results_per_nside(nside)):
+        ring_pixels = np.arange(tot_npix)
+        nest_pixels = healpy.ring2nest(nside, ring_pixels)
+        uniq_pixels = mhealpy.nest2uniq(nside, nest_pixels)
+        for uni, rin in zip(uniq_pixels, ring_pixels):
+            if uni not in uniq_list:
+                uniq_list.append(uni)
+                tmp_theta, tmp_phi = healpy.pix2ang(nside, rin)
+                tmp_dec = np.pi/2 - tmp_theta
+                tmp_ra = tmp_phi
+                grid_map[(tmp_dec, tmp_ra)] = np.nan
```

Comment:
You are right about this. In the probability maps the nan values are a problem in two cases:
1. In case of Gaussian convolution;
2. In the saved flattened map, as they could make the file very heavy.

I now changed the code so that the nan values are substituted only in one of these two cases, otherwise they're always preserved. Moreover, if the minimum of the map is bigger than 1e-16, then 1e-16 is chosen as filling value.

Reply:
So I understand 1. but not 2.

For 1. why not just set nans to 0, which is also the default of `np.nan_to_num`, then do the convolution.

For 2. I'm surprised that having `nans` don't offer the same level of compression for flattened maps. Does it really break the compression? I've not noticed this before with the LLH maps.